api/envoy/config/core/v3/protocol.proto

```diff
@@ -157,6 +157,10 @@ message HttpProtocolOptions {
   // If this setting is not specified, the value defaults to ALLOW.
   // Note: upstream responses are not affected by this setting.
   HeadersWithUnderscoresAction headers_with_underscores_action = 5;
+
+  // Optional maximum requests for both upstream and downstream connections.
+  // If not specified, there is no limit. It's a soft limit.
+  uint32 max_requests_per_connection = 6;
```

Comment:
Prefer `Uint32Value` to allow us to control defaults.

Reply:
We shouldn't expect the default to change away from no-limit. So, I prefer `uint32`

Reply:
I think in this case a default of "0" is ambiguous so I agree with @htuch I would make this a WKT.

Reply:
Changed to `Uint32Value`